ssl/ssl_privkey.cc

```diff
@@ -383,10 +383,25 @@ static bool ssl_public_key_rsa_pss_check(EVP_PKEY *pubkey, uint16_t sigalg) {
   return true;
 }
 
+static bool tls12_pkey_supports_cipher_auth(SSL_HANDSHAKE *hs,
+                                            const EVP_PKEY *key) {
+  SSL *const ssl = hs->ssl;
+  // We may have a private key that supports the signature algorithm, but we
+  // need to verify that the negotiated cipher allows it. This behavior is only
+  // done prior to TLS 1.2 servers in OpenSSL since TLS 1.3 does not have
+  // cipher-based authentication configuration. Since cipher-based
+  // authentication is already built into TLS 1.3, we use the |SSL_aGENERIC|
```

Comment:
this is a little confusing... suggest something like "Since authentication is configured outside the ciphersuite in TLSv1.3, ..."

Reply:
No problem at all, will change.